clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/OnDemandAgent.groovy

```diff
@@ -34,10 +34,26 @@ interface OnDemandAgent {
     Map<String, Collection<String>> evictions = [:]
   }
 
-  @Deprecated
-  boolean handles(String type)
+  enum OnDemandType {
+    ServerGroup,
+    SecurityGroup,
+    LoadBalancer,
+
+    static OnDemandType fromString(String s) {
+      if (s) {
```

Comment:
Good catch. I originally had declared them as SERVER_GROUP, so I had a "replace('_', '')" call in there.